api/api.go

```diff
@@ -13,29 +14,43 @@ import (
 	"github.com/jinzhu/gorm"
 	"github.com/netlify/netlify-commerce/conf"
 	"github.com/netlify/netlify-commerce/mailer"
+	"github.com/pborman/uuid"
 	"github.com/rs/cors"
-	"github.com/zenazn/goji/web/mutil"
 )
 
 var bearerRegexp = regexp.MustCompile(`^(?:B|b)earer (\S+$)`)
 
 // API is the main REST API
 type API struct {
 	handler    http.Handler
+	listener   net.Listener
```

Comment:
if you don't manage the listener you can't call shutdown on it. By managing it you can start the http server in a test and shut it down too.